### ISSUE TITLE
Data.Text.IO should not be used

### ISSUE THREAD

**Issue by Xitian9** [simonmichael/hledger#1619]
[It has been documented](https://www.snoyman.com/blog/2020/10/haskell-bad-parts-1/) that `Data.Text.IO` does not play well with locales, and can cause problems. We should reduce/eliminate its use in our code, explicitly encoding to utf8 before writing the resulting ByteString.

[More context:
<https://www.snoyman.com/blog/2016/12/beware-of-readfile>
<https://hledger.org/install.html#check-your-locale>
<https://hledger.org/1.25/hledger.html#troubleshooting>
Basically by addressing this with [#1834](https://github.com/simonmichael/hledger/pull/1834) we aim to once and for all solve or at least reduce "hledger crashed because $LANG was unset / wrong for this file / the locale was not installed / [Nix](https://github.com/simonmichael/hledger/issues/1033)"-type reports (see also issues linked on #1834). --SM]

**Comment by Xitian9** [comment]
This is complicated by the fact that we handle plain text files, which are written by an external program and so will be in your locale's character encoding. This leaves us forced to confront the problems that arise from this, rather than avoiding them by forcing utf-8 everwhere as the above article recommends.

**Comment by Xitian9** [comment]
In the instances where we do write files, do we use the locale or do we force utf-8?
